guides/CONTRIBUTING.md

```diff
@@ -18,26 +16,165 @@ When filing an issue, make sure to provide:
+
+### Fabric 
+
+Codegen introduced by the [new architecture](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/codegen.md) is fired automatically for iOS while runing `pod install` or when runing `FabricExample` in Android Studio. Developer can also run it manually by going into `./FabricExample/android` and running `./gradlew generateCodegenArtifactsFromSchema`. In both cases it is important to copy changed files from `/android/build/generated/source/codegen/java/com/facebook/react/viewmanagers/` to `android/src/paper/java/com/facebook/react/viewmanagers/`, so the interfaces are in sync for Paper architecture. 
+
+### Preparing Pull Request
+
+When your code changes are ready, it is time to open your Pull Request. GitHub already has a template that helps you properly post your changes. The most crucial are:
+
+1. **Description**:
+ - If you're solving specific issue, start with linking it.
+ - Write what are your motivations. 
+2. **Changes** - write what you have changes and why.
+3. **Screenshots / GIFs** &ndash; if applicable it's great idea to attach screen or video before and after changes.
+4. **Test code and steps to reproduce** &ndash; describe how others can test your change, if you didn't add `Test*.tsx` file it's good idea to add code snippets here. 
+
+### Do I need to prepare Pull Request for react-navigation too? 
+
+Currently, native stack is both in `react-native-screens` and `react-navigation`. `react-native-screens` contains native-stack v5 (`src/native-stack`), newer versions (v6, v7) are moved to `react-navigation` (available on branches `main`, `6.x`), hence in some cases it is necessary to prepare a pull request for `react-navigation` alongside the `react-native-screens` changes. 
+
+> [!CAUTION]
+> Currently, in this setup we have some code duplications that we're aware of. We're planning to deprecate native-stack v5 soon and remove it from react-native-screens.
+
+#### Changes in native code 
+
+If your change is only related to the native code (for example you're fixing a bug), you **do not need** to create a PR in the `react-navigation` repository. `react-native-screens` is a separate library installed alongside `react-navigation`, so you can just increase version of `react-native-screens` in your package.json file.
+
+#### Changes in JS code or both
+
+If you're changing native-stack v5 (`src/native-stack`) you **do not need** to create a PR in the `react-navigation` repository.
+
+If you're changing native-stack v6, v7 you **need to** create a PR in the `react-navigation` repository, since the code belongs there now.
 
-### Pull requests
+If you're changing core functionality:
+- If it's a bug fix and does not change the interface, then you **do not need** to create a PR in `react-navigation` repository.
+- If you're adding a new feature or changing API, you **need to** create a PR in `react-navigation` repository.
```

Comment:
```suggestion
- If you're adding a new feature or changing API, you **need to** create a PR in `react-navigation` repository exposing the feature there.
```

Reply:
Changed but rephrased a bit in https://github.com/software-mansion/react-native-screens/pull/2140/commits/b4ed2ce317a49caca2226c3a42bf56eacaf49088